libbeat/processors/add_cloud_metadata/provider_azure_vm.go

```diff
@@ -18,12 +18,46 @@
 package add_cloud_metadata
 
 import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v4"
+
 	s "github.com/elastic/beats/v7/libbeat/common/schema"
 	c "github.com/elastic/beats/v7/libbeat/common/schema/mapstriface"
 	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
 	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
+type azureMetadataFetcher struct {
+	provider               string
+	httpMetadataFetcher    *httpMetadataFetcher
+	genericMetadataFetcher *genericFetcher
+	httpMeta               mapstr.M
+}
+
+func newAzureMetadataFetcher(
+	provider string,
+	httpMetadataFetcher *httpMetadataFetcher,
+) (*azureMetadataFetcher, error) {
+
+	azFetcher := &azureMetadataFetcher{
+		provider:            provider,
+		httpMetadataFetcher: httpMetadataFetcher,
+	}
+	return azFetcher, nil
+}
+
+// NewClusterClient returns a NewManagedClustersClient
+var NewClusterClient func(clientFactory *armcontainerservice.ClientFactory) *armcontainerservice.ManagedClustersClient = func(clientFactory *armcontainerservice.ClientFactory) *armcontainerservice.ManagedClustersClient {
```

Comment:
The type should be omitted.

```suggestion
var NewClusterClient = func(clientFactory *armcontainerservice.ClientFactory) *armcontainerservice.ManagedClustersClient {
```